lib/http/request.rb

```diff
@@ -66,20 +66,24 @@ class UnsupportedSchemeError < RequestError; end
     # Scheme is normalized to be a lowercase symbol e.g. :http, :https
     attr_reader :scheme
 
+    attr_reader :uri_normalizer
+
     # "Request URI" as per RFC 2616
     # http://www.w3.org/Protocols/rfc2616/rfc2616-sec5.html
     attr_reader :uri
     attr_reader :proxy, :body, :version
 
     # @option opts [String] :version
     # @option opts [#to_s] :verb HTTP request method
+    # @option opts [#to_s] :uri_normalizer uri normalizer
     # @option opts [HTTP::URI, #to_s] :uri
     # @option opts [Hash] :headers
     # @option opts [Hash] :proxy
     # @option opts [String, Enumerable, IO, nil] :body
     def initialize(opts)
       @verb   = opts.fetch(:verb).to_s.downcase.to_sym
-      @uri    = normalize_uri(opts.fetch(:uri))
+      @uri_normalizer = opts[:normalize_uri] ? opts.fetch(:normalize_uri) : HTTP::URI::NORMALIZER
+      @uri = @uri_normalizer.call(opts.fetch(:uri))
```

Comment:
No need to assign normalizer here if it won't be used elsewhere. Also use fetch instead of ternary here:

``` ruby
normalize = opts.fetch(:normalize_uri, HTTP::URI::NORMALIZER)
@uri = normalize.call(opts.fetch(:uri))
```

Probably worth to verify if returned value is an HTTP::URI, as we might depend on it's API. Not sure on this count though.